2025-08-12 00:08:47,406 - __main__ - INFO - Starting training for fold 0
2025-08-12 00:08:50,069 - bin.genotype_dataset - INFO - Initialized dataset with 301 samples and 483 SNPs
Training Epoch 0:   0%|                                                                                                           | 0/10 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/Users/max/Desktop/PRS_Models/nextflow_pipeline_prs/bin/train_model.py", line 588, in <module>
    main()
  File "/Users/max/Desktop/PRS_Models/nextflow_pipeline_prs/bin/train_model.py", line 570, in main
    trainer.train(args.max_epochs)
  File "/Users/max/Desktop/PRS_Models/nextflow_pipeline_prs/bin/train_model.py", line 342, in train
    train_loss, train_metrics = self.train_epoch()
  File "/Users/max/Desktop/PRS_Models/nextflow_pipeline_prs/bin/train_model.py", line 199, in train_epoch
    for batch_idx, (genotypes, phenotypes) in enumerate(pbar):
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/site-packages/tqdm/std.py", line 1181, in __iter__
    for obj in iterable:
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 489, in __iter__
    self._iterator = self._get_iterator()
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 427, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 1172, in __init__
    w.start()
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/context.py", line 288, in _Popen
    return Popen(process_obj)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
  File "/Users/max/miniconda3/envs/prs_methods/lib/python3.10/site-packages/h5py/_hl/base.py", line 369, in __getnewargs__
    raise TypeError("h5py objects cannot be pickled")
TypeError: h5py objects cannot be pickled
